lib/libshare/libshare_impl.h

```diff
@@ -22,36 +22,36 @@
 /*
  * Copyright (c) 2002, 2010, Oracle and/or its affiliates. All rights reserved.
  * Copyright (c) 2011 Gunnar Beutner
+ * Copyright (c) 2019, 2020 by Delphix. All rights reserved.
  */
 
-struct sa_handle_impl;
-
 typedef struct sa_share_fsinfo {
-	boolean_t active;
-	char *resource;
 	char *shareopts;
 } sa_share_fsinfo_t;
 
 typedef struct sa_share_impl {
-	struct sa_share_impl *next;
-
-	struct sa_handle_impl *handle;
-
-	char *sharepath;
-	char *dataset;
+	char *sa_mountpoint;
+	char *sa_zfsname;
 
 	sa_share_fsinfo_t *fsinfo; /* per-fstype information */
 } *sa_share_impl_t;
 
 #define	FSINFO(impl_share, fstype) (&(impl_share->fsinfo[fstype->fsinfo_index]))
 
+#define	LIBSHARE_LOCK_FILE	"/etc/zfs/libshare.lck"
```

Comment:
is `.lck` a pre-existing convention?  If not, I think spelling out `.lock` would be helpful.